src/gui/macOS/fileprovidersocketserver.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2022 by Claudio Cambra <claudio.cambra@nextcloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#pragma once
+
+#include <QObject>
+#include <QLocalServer>
+#include <QLoggingCategory>
+
+namespace OCC {
+
+namespace Mac {
+
+class FileProviderSocketController;
+using FileProviderSocketControllerPtr = QSharedPointer<FileProviderSocketController>;
+
+#ifdef Q_OS_MACOS
+QString fileProviderSocketPath();
+#endif
+
+class FileProviderSocketServer : public QObject
+{
+    Q_OBJECT
+
+public:
+    explicit FileProviderSocketServer(QObject *parent = nullptr);
+
+private slots:
+    void startListening();
+    void slotNewConnection();
+    void slotSocketDestroyed(const QLocalSocket * const socket);
+
+private:
+    QString _socketPath;
+    QLocalServer _socketServer;
+    QHash<const QLocalSocket*, FileProviderSocketControllerPtr> _socketControllers;
```

Comment:
why is this using a `QShqredPointer` ?

Reply:
my question is really do you need to count more than one reference to the allocated object ? if you only have one pointer tracking it, `QSharedPointer` is not the good solution

Reply:
Okay, changed